test/integration/puppetlabs/services/master/environment_transports_int_test.clj

```diff
@@ -0,0 +1,638 @@
+             (is (= initial-etag (response-etag response))
+                 "etag changed even though code did not")
+             (is (= expected-initial-response
+                    (response->map response))
+                 "unexpected body for response")))
+         (testing (str "HTTP 304 (not modified) returned when request "
+                       "roundtrips last etag and code has not changed")
+           (let [response (get-env-transports "production" initial-etag)]
+             (is (= 304 (:status response))
+                 (str
+                  "unexpected status code for response for no code change and "
+                  "original etag roundtripped"))
+             (is (= initial-etag-without-gzip-suffix (response-etag response))
+                 "etag changed even though code did not")
+             (is (empty? (:body response))
+                 "unexpected body for response")))
+         (testing (str "SERVER-1153 - HTTP 304 (not modified) returned when "
+                       "request roundtrips last etag with '--gzip' suffix and "
+                       "code has not changed")
+           (let [response (get-env-transports "production"
+                                           initial-etag-with-gzip-suffix)]
+             (is (= 304 (:status response))
+                 (str
+                  "unexpected status code for response for no code change and "
+                  "etag with '--gzip' suffix roundtripped"))
+             (is (= initial-etag-without-gzip-suffix (response-etag response))
+                 "etag changed even though code did not")
+             (is (empty? (:body response))
+                 "unexpected body for response")))
+         (comment
+           "Not valid until we write new data out"
+           (testing (str "environment_transport fetch without if-none-match "
+                         "header includes latest info after code update")
+             (str "test data")
+             (str "test data")
+             (let [baz-file (str "test data")
+                   borked-file (str "test data")
+                   _ (purge-all-env-caches)
+                   response (get-env-transports "production")]
+               (is (= 200 (:status response))
+                   (str
+                    "unexpected status code for response following code change,"
+                    "response: "
+                    (ks/pprint-to-string response)))
+               (is (not= initial-etag (response-etag response))
+                   "etag did not change even though code did")
+               (is (= mock-response
+                      (response->map response))
+                   "unexpected body following code change")))))
+       (catch Exception e
+         (println "dumping puppetserver.log\n" (slurp debug-log))
+         (throw e))))))
+
+
+#_(deftest ^:integration environment-transports-integration-cache-enabled-test3
```

Comment:
Does this mean this test is skipped right now?

Reply:
yeah, `#_` is a reader macro that comments out the following form. this whole deftest is handling updating caches after changing resources on disk, which isn't valid until we get rid of the stub ruby impl.